events.md

```diff
@@ -0,0 +1,271 @@
+4. To a specific skill installed at an organization
+    1. Used for CRUD events that apply to an entire org, e.g. `booking:create-service`
+    2. The event name starts with the target skill's slug
+
+## Step 1: Event contracts
+
+Just like you did for events you want to subscribe to, you have to define the skills you will be emitting.
+
+### Example
+
+```js
+// config/default.js
+eventContract: {
+  events: {
+    'my-awesome-skill:did-trigger-motion-sensor': { // the slug of your skill and then the name of your event
+        publish: true // you are publishing this event
+    },
+    'my-awesome-skill:set-motion-sensor-sensitivity': {
+        publish: true
+    }
+  }
+}
+```
+
+## Step 2: Emit your event
+
+### Example
+
+```js
+// emit your custom will- event to all skills enabled at a location
+const willResponses = await ctx.sb.emit(
+    locationId,
+    "my-awesome-skill:will-do-something",
+    {
+        foo: "bar"
+    }
+);
+
+// example on how to emit to all skills installed at an org (with no location)
+/*
+const willResponses = await ctx.sb.emitOrganization(
+    organizationId,
+    "my-awesome-skill:will-do-something",
+    {
+        foo: "bar"
+    }
+);
+*/
+
+// did anyone prevent default?
+const preventDefault = willResponses.reduce((preventDefault, response) => {
+    return preventDefault || !!response.payload.preventDefault;
+}, false);
+
+// if someone prevented default, bail
+if (preventDefault) {
+    return false;
+}
+
+// we were not canceled, lets look at the first response and see if the skill passed anything back
+const first = willResponses[0];
+let results = [];
+
+if (first && first.payload.someRandomOption) {
+    log.debug(
+        `the first skill responded with someRandomOption = ${someRandomOption}`
+    );
+
+    // Emit to a CRUD event in another skill and wait for the results
+    results = await ctx.sb.emit(
+        locationId,
+        "another-skill:create-new-records",
+        {
+            someRandomOption: true
+        }
+    );
+}
+
+// now that we're done, lets let everyone know
+await ctx.sb.emit(locationId, "my-awesome-skill:did-do-something", {
+    hello: "again",
+    results: results
+});
+```
+
+## Emit API
+
+-   `ctx.sb.emit(locationId, eventName, payload = {}, options, eventId)`
+-   `ctx.sb.emitOrganization(organizationId, eventName, payload = {}, options, eventId)`
+
+Attributes:
+
+-   `locationId`: `UUID4` - The id of the location you want to emit to.
+-   `eventName`: `string` - The name of the event you are emitting. If the name starts with your skill`s slug, the event will go to every skill installed at the location. If the name starts with another skill's slug, it'll only go to that skill.
+-   `payload`: `object` (optional) - Anything you can `JSON.stringify` that is passed along with the event.
+-   `options` : `object` (optional) - For future use.
+-   `eventId`: `UUID4` - pass a custom UUID that is passed through to all skills listening to the event
```

Comment:
`eventId` is optional. If it is not passed, an eventId is generated

`options` has parameters in use currently:
`options.timeout` (integer) - the timeout in ms (default 5000)
`options.retry` (boolean) - whether the event should be retried on failure (default true)